Update CHANGELOG.md for main:2025-05-11 (#18168)
ydb-platform/ydb: CHANGELOG.md

* 18121:`ALTER TABLE table ` should not fail for a table with vector index. [#18121](https://github.com/ydb-platform/ydb/pull/18121) ([azevaykin](https://github.com/azevaykin))
* 18088:Don't allow observing inconsistent results in some read-write transactions. Fixes #18064. [#18088](https://github.com/ydb-platform/ydb/pull/18088) ([Aleksei Borzenkov](https://github.com/snaury))
* 18059:Fixed crashes by some SELECT queries with DESC in OLAP tables [#18059](https://github.com/ydb-platform/ydb/pull/18059) ([Semyon](https://github.com/swalrus1))
* 17982:`ALTER TABLE table RENAME INDEX` should not fail for a vector index. [#17982](https://github.com/ydb-platform/ydb/pull/17982) ([azevaykin](https://github.com/azevaykin))
* 17814:Fixed BuildIndex/Export/Import operation listing order (#17817) [#17814](https://github.com/ydb-platform/ydb/pull/17814) ([Vitaliy Filippov](https://github.com/vitalif))
* 17729:Fixed schema version collisions in serverless DBs. #17184 [#17729](https://github.com/ydb-platform/ydb/pull/17729) ([Semyon](https://github.com/swalrus1))

### YDB UI

> * 17942:viewer: change auth for `whoami` and `capabilities` handlers [#17942](https://github.com/ydb-platform/ydb/pull/17942) ([Andrei Rykov](https://github.com/StekPerepolnen))

Reviewer: ```suggestion
* 17942:Changed authentication mechanisms for `whoami` and `capabilities` handlers in the viewer. [#17942](https://github.com/ydb-platform/ydb/pull/17942) ([Andrei Rykov](https://github.com/StekPerepolnen))
```

### Performance

> * 17611:Introduce Intersect boolean operation to Roaring UDF. [#17611](https://github.com/ydb-platform/ydb/pull/17611) ([jsjant](https://github.com/jsjant))

Reviewer: ```suggestion
* 17611:Introduced the Intersect boolean operation to Roaring UDF. [#17611](https://github.com/ydb-platform/ydb/pull/17611) ([jsjant](https://github.com/jsjant))
```

> * 17533:Constant folding now also handles deterministic UDFs. Non-deterministic UDFs are detected via a black list [#17533](https://github.com/ydb-platform/ydb/pull/17533) ([Pavel Velikhov](https://github.com/pavelvelikhov))

Reviewer: ```suggestion
* 17533:Extended constant folding to include deterministic UDFs, while non-deterministic UDFs are identified through a blacklist. [#17533](https://github.com/ydb-platform/ydb/pull/17533) ([Pavel Velikhov](https://github.com/pavelvelikhov))
```
